cmd/query/app/server_test.go

```diff
@@ -855,3 +859,104 @@ func TestServerHTTPTenancy(t *testing.T) {
 		})
 	}
 }
+
+func TestServerHTTP_TracesRequest(t *testing.T) {
```

Comment:
This is a great test, but it highlights a problem. I recommend creating a separate PR with just this test, and it will fail on main, which means we are changing something about the behavior.

Reply:
@yurishkuro this is rebased now